pkg/controller/deployment/deployment_controller.go

```diff
@@ -242,6 +276,33 @@ func (d *DeploymentController) reconcileOldRCs(allRCs []*api.ReplicationControll
 	return true, err
 }
 
+func (d *DeploymentController) scaleDownOldRCs(oldRCs []*api.ReplicationController, deployment extensions.Deployment) (bool, error) {
```

Comment:
Add a comment noting that this method is intended to be used with the recreate strategy.